### PR TITLE
🐛 [amp-next-page 1.0] Allow shadowdoc children to utilize Viewer CID API

### LINKED ISSUE #32305: [amp-next-page] AMP Client ID changes when scrolling through to amp-next-page in AMP Viewer
## What's the issue?

The [AMP Client ID](https://webcache.googleusercontent.com/search?q=cache%3AVRskkkr1q5IJ%3Ahttps%3A%2F%2Fdevelopers.google.com%2Fanalytics%2Fdevguides%2Fcollection%2Famp-analytics%2Fclient-id%20&cd=1&hl=en&ct=clnk&gl=us) (used in amp-analytics) is changing on scrolling into amp-next-page in an amp viewer (but not on publisher origin).

## How do we reproduce the issue?

Please provide a public URL and ideally a reduced test case (e.g. on jsbin.com) that exhibits only your issue and nothing else. Provide step-by-step instructions for reproducing the issue:

1. Go to https://www.google.com/amp/s/www.cnet.com/google-amp/news/nasa-curiosity-rover-snaps-dramatic-mars-selfie-at-scenic-mont-mercou/ in mobile emulator with network tab open in Devtools.
2. Scroll through and into the second article.
3. Search for "v56=&AQE=1" in the network tab. You should see two records. Compare the vid query params for each of them. They are distinct even though they are for the same session. (If you open another tab and go to, for example, https://www.google.com/amp/s/www.cnet.com/google-amp/news/nasa-curiosity-rover-snaps-dramatic-mars-selfie-at-scenic-mont-mercou/ and scroll through, you will see that the client id of the first article will correspond to that of the first article of the previous stream you looked at, but the second article won't)

## What browsers are affected?

Viewer only. Chrome

## Which AMP version is affected?

2101211748002

(cc @dmanek)

### PR DESCRIPTION
**Pull request by micajuine-ho**
Closes #32305

When server in a viewer, if the Viewer CID API is supported, then we ask the Viewer to get/generate a CID of us (per domain per user). 

We know that the Viewer CID API is supported if the ampdoc has the param `cap` with `cid` inside as a value. 
https://github.com/ampproject/amphtml/blob/99c70c40ae92a247783d3e24eca7eeea0d1f75f6/src/service/viewer-cid-api.js#L44-L53

amp-next-page creates shadow docs for its children. If it's served in a viewer, then the host pages' ampdoc will have the param `cap` with `cid` somewhere in it's value. 

It makes sense that we should also add these capabilities to the children shadowdoc's ampdocs. This way the shadow docs will also know that the Viewer CID API is supported.

Once the shadowdoc knows that the Viewer CID API is. supported, it will send a message to the viewer, all the other fields in the message are already provided to the shadowdoc and should work: https://github.com/ampproject/amphtml/blob/main/src/service/viewer-cid-api.js#L55-L72

### CODE REVIEW

**Review by dmanek — APPROVED** [review]

**Review by jridgewell — CHANGES_REQUESTED** [review]
Can multidoc handle all of the [capabilities](https://github.com/ampproject/amphtml/blob/main/extensions/amp-viewer-integration/CAPABILITIES.md)?

**Comment by micajuine-ho** [comment]
@jridgewell Thanks for linking the documentation. I would be hesitant to assume that multidoc can handle all of the capabilities. So, I think we should only pass in CID capabilities if it is there.

File: extensions/amp-next-page/1.0/service.js [hunk 1]
```diff
@@ -552,6 +552,20 @@ export class NextPageService {
...
+   * Forward the allowlisted capabilities from the viewer
+   * to the multidoc's ampdocs (i.e. CID), so they know
+   * the viewer supports it for the multidoc.
+   * @return {?string}
+   */
+  getCapabilities_() {
+    const viewerCapabilities = this.ampdoc_.getParam('cap');
+    if (viewerCapabilities && viewerCapabilities.indexOf('cid') > -1) {
+      return 'cid';
+    }
```
Comment (jridgewell) [inline extensions/amp-next-page/1.0/service.js:565]: Please use `viewer.hasCapibility`

**Review by jridgewell — APPROVED** [review]

**Pull request merged**